include/up-client-vsomeip-cpp/routing/SomeipRouter.hpp

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SOMEIP_ROUTER_HPP
+#define SOMEIP_ROUTER_HPP
+#include <up-client-vsomeip-cpp/routing/SomeipHandler.hpp>
+#include <up-cpp/transport/UTransport.h>
+#include "SomeipRouterInterface.hpp"
+#include <unordered_map>
+
+/**
+ * @brief Implements the SomeipRouter functionality.
```

Comment:
How is the router different from the handler?

Reply:
I can explain the overall design more, if you need me to, but to explain in brief:
Handler class is the service handler, every service (UEId/ServiceId) has a dedicated handler class instance that handles incoming and outgoing messages. SomeipRouter is the bookkeeper for all Handler instances and knows which handler to route the outgoing message to based on ueid/serviceid.